MusicCat-Console/Program.cs

```diff
@@ -52,5 +64,40 @@ static void Main(string[] args)
 			logStream?.Dispose();
             Listener.Stop();
         }
+
+		private static int counter;
+		private static void OnConfigChange(object sender, FileSystemEventArgs e)
+		{
+			if (counter == 0)
+			{
+				counter++;
+				return;
+			}
+
+			counter = 0;
+			try
+			{
+				logWriter?.Dispose();
+				logStream?.Dispose();
+				Listener.Config = Config.ParseConfig(out logStream, out logWriter);
+			}
+			catch (Exception ex)
+			{
+				Log(new ApiLogMessage($"Failed to de-serialize config, using default config instead. Exception: {ex.Message}{Environment.NewLine}{ex.StackTrace}", ApiLogLevel.Warning));
+				Listener.Config = Config.DefaultConfig;
+			}
+			Listener.Stop();
+			Listener.Start();
+			try
+			{
+				MetadataStore.SongList.Clear();
+				MetadataStore.LoadMetadata(Log);
```

Comment:
I would prefer to have the metadata store load and validate the incoming metadata before discarding the old metadata. That way if the new metadata is invalid, it can drop it and stick with the old set.